src/main/java/com/philips/swcoe/cerberus/cerebellum/codemetrics/java/CodeMetricsService.java

```diff
@@ -7,16 +7,36 @@
 import com.github.mauricioaniche.ck.CK;
 import com.github.mauricioaniche.ck.CKClassResult;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 
 public class CodeMetricsService {
 
-    public static List<CKClassResult> getCodeMetrics(String pathToSource) {
+    public static List<CKClassResult> getCodeMetrics(String pathToSource, String exclusionFiles) {
         List<CKClassResult> codeStatistics = new ArrayList<>();
         new CK().calculate(pathToSource, false, result -> {
             codeStatistics.add(result);
         });
-        return codeStatistics;
+        if (exclusionFiles != null) {
+            return doExclusion(exclusionFiles, codeStatistics);
+        } else {
+            return codeStatistics;
+        }
+    }
+
+    private static List<CKClassResult> doExclusion(String exclusionFiles, List<CKClassResult> codeStatistics) {
+        List<CKClassResult> codeStatisticsWithExclusion = new ArrayList<>();
+
+        List<String> exclusionLists = Arrays.asList(exclusionFiles.split(","));
+        for (CKClassResult codeStatistic : codeStatistics) {
+            for (String exclusionList : exclusionLists) {
+                if (!codeStatistic.getFile().contains(exclusionList)) {
+                    codeStatisticsWithExclusion.add(codeStatistic);
+                }
+            }
```

Comment:
Can you please re-visit the logic here, you are looping exclusionlist every time .